account_invoice_refund_reason/wizard/account_move_reversal.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (C) 2019 Open Source Integrators
+# Copyright (C) 2019 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMoveReversal(models.TransientModel):
+    _inherit = "account.move.reversal"
+
+    reason_id = fields.Many2one(
+        "account.invoice.refund.reason", string="Reason to credit"
+    )
+
+    @api.onchange("reason_id")
+    def _onchange_reason_id(self):
+        if self.reason_id:
+            self.reason = self.reason_id.name
+
+    def reverse_moves(self):
+        res = super().reverse_moves()
+        inv_obj = self.env["account.move"]
+        context = dict(self._context or {})
+        for inv in inv_obj.browse(context.get("active_ids")):
+            inv.reason_id = self.reason_id.id
+        return res
```

Comment:
Wouldn't it make more sense to log the refund reason on the credit note?

You could have multiple refunds for one invoice, so I think it would make more sense to do it like this.